exporter/awsemfexporter/pusher.go

```diff
@@ -24,60 +26,84 @@ import (
 )
 
 const (
-	//http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html
-	//In truncation logic, it assuming this constant value is larger than PerEventHeaderBytes + len(TruncatedSuffix)
-	MaxEventPayloadBytes = 1024 * 256 //256KB
+	// http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html
+	// In truncation logic, it assuming this constant value is larger than PerEventHeaderBytes + len(TruncatedSuffix)
+	DefaultMaxEventPayloadBytes = 1024 * 256 //256KB
 	// http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
 	MaxRequestEventCount   = 10000
 	PerEventHeaderBytes    = 26
 	MaxRequestPayloadBytes = 1024 * 1024 * 1
 
-	logEventChanBufferSize    = 10000 // 1 request can handle max 10000 log entries
-	minPusherIntervalInMillis = 200   // 5 TPS
+	minPusherIntervalInMillis = 200 // 5 TPS
 
-	logEventBatchPushChanBufferSize = 2 // processing part does not need to be blocked by the current put log event request
-	TruncatedSuffix                 = "[Truncated...]"
+	TruncatedSuffix = "[Truncated...]"
 
 	LogEventTimestampLimitInPast   = 14 * 24 * time.Hour //None of the log events in the batch can be older than 14 days
 	LogEventTimestampLimitInFuture = -2 * time.Hour      //None of the log events in the batch can be more than 2 hours in the future.
 )
 
-//Struct to present a log event.
+var (
+	maxEventPayloadBytes = DefaultMaxEventPayloadBytes
+)
+
+// Struct to present a log event.
 type LogEvent struct {
 	InputLogEvent *cloudwatchlogs.InputLogEvent
-	//The time which log generated.
+	// The time which log generated.
 	LogGeneratedTime time.Time
 }
 
-//Calculate the log event payload bytes.
-func (logEvent *LogEvent) eventPayloadBytes() int {
-	return len(*logEvent.InputLogEvent.Message) + PerEventHeaderBytes
+// Create a new log event
+// logType will be propagated to logEventBatch and used by pusher to determine which client to call PutLogEvent
+func NewLogEvent(timestampInMillis int64, message string) *LogEvent {
```

Comment:
Keep methods private unless they must be public (generally only the ones that implement an interface)

Reply:
We generally use `timestampMS` convention

Reply:
Thanks for the suggestion. All fixed.